### PR TITLE
chore: migrates to vitejs

### LINKED ISSUE #796: migrate to vite
### Is your feature request related to a problem? Please describe

vue-cli is really slow while developing and many packages have been discontinued.

### Describe the solution you'd like

switch from vue-cli to vite.

### Describe alternatives you've considered

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by pedrolamas**
## Current status

(tasks might change as I come by other things)

- [x] Internal file editor (monaco editor)
- [X] G-Code Previewer
- [X] Styles
- [X] Tests
- [x] GitHub workflows
- [x] Eslint
- [x] Dynamic favicon
- [x] PWA
- [x] Version

## Notes

- **Build time goes from ~3m40s to ~60s!!!**
- vue-class-component does NOT make class properties that are initialized as `undefined` as reactive, and we have quite a few of those in our code (though I have found and replaced a few already on this work). More info on this [here](https://class-component.vuejs.org/guide/class-component.html#data).
- I changed the tsconfig.json target to es2020, this defaults `useDefineForClassFields: true` and that created quite a few issues (linked to the above, but also other types of bindings failing). For now, I've forced `useDefineForClassFields: false`, but we SHOULD (MUST?) revert this when we migrate to Vue 3.

Fixes #796

Signed-off-by: Pedro Lamas <pedrolamas@gmail.com>

### CODE REVIEW

**Comment by pedrolamas** [comment]
I just had this issue:

https://github.com/vitejs/vite/issues/2433

Had to build with `NODE_OPTIONS='--max-old-space-size=4096' vite build` to get the source maps working...

File: src/components/widgets/filesystem/setupMonaco.ts [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
+import 'monaco-editor/esm/vs/editor/standalone/browser/quickAccess/standaloneGotoLineQuickAccess.js'
+// import 'monaco-editor/esm/vs/editor/standalone/browser/quickAccess/standaloneGotoSymbolQuickAccess.js'
+import 'monaco-editor/esm/vs/editor/standalone/browser/quickAccess/standaloneCommandsQuickAccess.js'
+// import 'monaco-editor/esm/vs/editor/standalone/browser/referenceSearch/standaloneReferenceSearch.js'
+
+// import 'monaco-editor/esm/vs/language/typescript/monaco.contribution'
+import 'monaco-editor/esm/vs/language/css/monaco.contribution'
+import 'monaco-editor/esm/vs/language/json/monaco.contribution'
+// import 'monaco-editor/esm/vs/language/html/monaco.contribution'
+// import 'monaco-editor/esm/vs/basic-languages/monaco.contribution'
```
Comment (matmen) [inline src/components/widgets/filesystem/setupMonaco.ts:16]: Are the commented out imports here intended?

**Review by matmen — CHANGES_REQUESTED** [review]
Nice! Functionality wise this seems to work almost perfectly (and builds much faster, too!!)

I did notice that the panning behavior of the bed mesh chart seems weird now, but I didn't see any code-related changes for that..

Also spotted a small typo:

File: src/components/widgets/filesystem/setupMonaco.ts (hunk 1 above)
Comment (pedrolamas) [reply src/components/widgets/filesystem/setupMonaco.ts:16]: I left those as reference, but thinking again, maybe I'll just add a link to the list of possible imports instead.

Comment (pedrolamas) [reply src/components/widgets/filesystem/setupMonaco.ts]: All clear now!

**Comment by pedrolamas** [comment]
> I did notice that the panning behavior of the bed mesh chart seems weird now, but I didn't see any code-related changes for that..

I haven't noticed anything... how can I repro this issue?

**Comment by matmen** [comment]
> > I did notice that the panning behavior of the bed mesh chart seems weird now, but I didn't see any code-related changes for that..
> 
> I haven't noticed anything... how can I repro this issue?

Hmm, not sure how to reproduce it. In the 1.21.1 prod build it seems like the the chart follows the cursor much more closely while panning, now it seems to only pan at about half the rate the cursor is moving. I'll take a look at it again later..

**Comment by matmen** [comment]
> > > I did notice that the panning behavior of the bed mesh chart seems weird now, but I didn't see any code-related changes for that..
> > 
> > 
> > I haven't noticed anything... how can I repro this issue?
> 
> Hmm, not sure how to reproduce it. In the 1.21.1 prod build it seems like the the chart follows the cursor much more closely while panning, now it seems to only pan at about half the rate the cursor is moving. I'll take a look at it again later..

This seems to have magically fixed itself after a rebuild, not sure what happened there..

File: index.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width,initial-scale=1.0">
+    <title>Fluidd</title>
```
Comment (matmen) [inline index.html:7]:
```suggestion
    <title>fluidd</title>
```

File: index.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    <meta name="description" content="The Klipper web interface for managing your 3d printer" />
+
+    <link rel="icon" type="image/png" sizes="32x32" href="/img/icons/favicon-32x32.png" />
+    <link rel="icon" type="image/png" sizes="16x16" href="/img/icons/favicon-16x16.png" />
+    <meta name="theme-color" content="#2196F3" />
+    <meta name="mobile-web-app-capable" content="yes" />
+    <meta name="apple-mobile-web-app-capable" content="yes" />
+    <meta name="apple-mobile-web-app-status-bar-style" content="black" />
+    <meta name="apple-mobile-web-app-title" content="Fluidd" />
```
Comment (matmen) [inline index.html:16]:
```suggestion
    <meta name="apple-mobile-web-app-title" content="fluidd" />
```

File: index.html [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    <link rel="apple-touch-icon" sizes="180x180" href="/img/icons/apple-touch-icon-180x180.png" />
+    <link rel="mask-icon" href="img/icons/safari-pinned-tab.svg" color="#000000" />
+    <meta name="msapplication-TileImage" content="/img/icons/mstile-150x150.png" />
+    <meta name="msapplication-TileColor" content="#000000" />
+
+    <script type="module" src="/src/main.ts"></script>
+  </head>
+  <body>
+    <noscript>
+      <strong>We're sorry but Fluidd doesn't work properly without JavaScript enabled. Please enable it to continue.</strong>
```
Comment (matmen) [inline index.html:26]: Not sure if it makes sense to change this one, as we're using capitalized Fluidd in the readme too:
```suggestion
      <strong>We're sorry, but fluidd doesn't work properly without JavaScript enabled. Please enable it to continue.</strong>
```

File: vite.config.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+import version from './src/plugins/vite-plugin-inject-version'
+
+export default defineConfig({
+  plugins: [
+    VitePWA({
+      registerType: 'autoUpdate',
+      includeAssets: ['img/**/*.svg', 'img/**/*.png'],
+      manifest: {
+        name: 'Fluidd',
+        short_name: 'Fluidd',
```
Comment (matmen) [inline vite.config.ts:21]:
```suggestion
        name: 'fluidd',
        short_name: 'fluidd',
```

**Review by matmen — DISMISSED** [review]
LGTM!!

I did notice that, when loading, the title is capitalized, whereas after initialization it's lowercased; it seemed weird to me but either way it's fine with me:

**Pull request merged**